test/asynchronous/test_monitor.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2014-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test the monitor module."""
+from __future__ import annotations
+
+import asyncio
+import gc
+import subprocess
+import sys
+import warnings
+from functools import partial
+
+sys.path[0:0] = [""]
+
+from test.asynchronous import AsyncIntegrationTest, async_client_context, connected, unittest
+from test.utils import (
+    ServerAndTopologyEventListener,
+    async_wait_until,
+)
+
+from pymongo.periodic_executor import _EXECUTORS
+
+_IS_SYNC = False
+
+
+def unregistered(ref):
+    gc.collect()
+    return ref not in _EXECUTORS
+
+
+def get_executors(client):
+    executors = []
+    for server in client._topology._servers.values():
+        executors.append(server._monitor._executor)
+        executors.append(server._monitor._rtt_monitor._executor)
+    executors.append(client._kill_cursors_executor)
+    executors.append(client._topology._Topology__events_executor)
+    return [e for e in executors if e is not None]
+
+
+class TestMonitor(AsyncIntegrationTest):
+    async def create_client(self):
+        listener = ServerAndTopologyEventListener()
+        client = await self.unmanaged_async_single_client(event_listeners=[listener])
```

Comment:
```suggestion
        client = await self.async_single_client(event_listeners=[listener])
```

Avoid using the `unmanaged` helper methods wherever possible.

Reply:
looking at the tests, i thought it was intentional to use unmanaged because the tests want to explicitly close the client and test behavior after client close (or client del)?

Reply:
Does it cause failures to switch the call from unmanaged?

Reply:
Yes we have to used unmanaged here, otherwise the client will never get garbage collected.

Reply:
Adding client.close to the test cleanup will keep a reference to the client and prevent it from being GCd.

Reply:
Ah, the reference inside the cleanups list will cause the `del client` call to not GC the client?

Reply:
Oh wait: a GC'd AsyncMongoClient will trigger the unclosed client warning.

Reply:
We still want to make sure that a GC'd async client's tasks eventually get cleaned up.